src/main.cpp

```diff
@@ -5183,7 +5187,7 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
             LOCK(cs_main);
             State(pfrom->GetId())->fProvidesHeaderAndIDs = !(nLocalServices & NODE_WITNESS) || (nCMPCTBLOCKVersion == 2);
             State(pfrom->GetId())->fPreferHeaderAndIDs = fAnnounceUsingCMPCTBLOCK;
-            State(pfrom->GetId())->fWantCmpctWitness = (nCMPCTBLOCKVersion == 2);
+            State(pfrom->GetId())->fSupportsCmpctWitness = (nCMPCTBLOCKVersion == 2);
```

Comment:
@sdaftuar I think they have to be sent in increasing order, so that the
receiver can use the simple "the last understood sendcmpct counts" logic.

You are right that we should ignore v2 sendcmpct messages when we don't
have NODE_WITNESS. That would make the behaviour of a non-defined segwit
node identical to a pre-segwit node. Agree?